geonode/maps/urls.py

```diff
@@ -49,6 +49,7 @@
     url(r'^new/data$', 'new_map_json', name='new_map_json'),
     url(r'^checkurl/?$', 'ajax_url_lookup'),
     url(r'^snapshot/create/?$', 'snapshot_create'),
+    url(r'^new/copy/(?P<mapid>[^/]+)$', existing_map_view, name='copy_map'),
```

Comment:
these statements separate the template for the two backends:
```
if 'geonode.geoserver' in settings.INSTALLED_APPS:
    new_map_view = 'new_map'
    existing_map_view = 'map_view'

elif 'geonode.qgis_server' in settings.INSTALLED_APPS:
    new_map_view = MapCreateView.as_view()
    existing_map_view = MapDetailView.as_view()
```

so, i'll create two new views for handling the two backends.